### ISSUE TITLE
MATCHES_IN_GROUP should work with a number

### ISSUE THREAD

**Issue by ivakegg** [NationalSecurityAgency/datawave#2400]
If you do something like $MATCHES__IN_GROUP(field1,value1,field2,10) where field2 is typed as a number, and the value is a number (10 in this case), then the query will fail as follows:

Caused by : ...JexlException: ...JexlEvaluation... (grouping:matchesInGroup(field1,value1,field2,+bE1) method invocation error
Caused by: java.util.regex.PatternSyntaxException: Dangling meta charater '+' nead index 0 +bE1

Basically, we need to normalize the values into a regex.

This may bring up a larger issue of when a user supplies a raw number where a regex is required, we need to think about how to handle that case.
